tests/integ/scala/test_function_suite.py

```diff
@@ -4271,6 +4275,449 @@ def test_to_array(session):
         sort=False,
     )
 
+    actual = session.createDataFrame(
+        [
+            ("Eva", 4, 2.5, 2, 5, 2.1, 6.3),
+            ("Alice", 2, 2.5, 2, 5, 2.0, 5.0),
+            ("Bob", 5, 5.5, 2, 5, 2.0, 5.0),
+        ],
+        [
+            "name",
+            "age",
+            "double_value",
+            "LITERAL_2_INT",
+            "LITERAL_5_INT",
+            "LITERAL_2_FLOAT",
+            "LITERAL_5_FLOAT",
+        ],
+    )
+    actual = actual.select(to_array("age", "age"))
+
+    Utils.check_answer(
+        actual,
+        [
+            Row("[\n  4,\n  4\n]"),
+            Row("[\n  2,\n  2\n]"),
+            Row("[\n  5,\n  5\n]"),
+        ],
+    )
+
+    actual = session.createDataFrame(
+        [
+            ("Eva", 4, 2.5, 2, 5, 2.1, 6.3),
+            ("Alice", 2, 2.5, 2, 5, 2.0, 5.0),
+            ("Bob", 5, 5.5, 2, 5, 2.0, 5.0),
+        ],
+        [
+            "name",
+            "age",
+            "double_value",
+            "LITERAL_2_INT",
+            "LITERAL_5_INT",
+            "LITERAL_2_FLOAT",
+            "LITERAL_5_FLOAT",
+        ],
+    )
```

Comment:
you don't have to recreate the same dataframe for each test. Just do 

```
    df0 = session.createDataFrame(
        [
            ("Eva", 4, 2.5, 2, 5, 2.1, 6.3),
            ("Alice", 2, 2.5, 2, 5, 2.0, 5.0),
            ("Bob", 5, 5.5, 2, 5, 2.0, 5.0),
        ],
        [
            "name",
            "age",
            "double_value",
            "LITERAL_2_INT",
            "LITERAL_5_INT",
            "LITERAL_2_FLOAT",
            "LITERAL_5_FLOAT",
        ],
    )
    expected = [
        Row("[\n  4,\n  4\n]"),
        Row("[\n  2,\n  2\n]"),
        Row("[\n  5,\n  5\n]"),
    ]
    Utils.check_answer(df0.select(to_array(df0.age, df0.age)), expected)
    Utils.check_answer(df0.select(to_array([df0.age, df0.age])), expected)
    Utils.check_answer(df0.select(to_array(df0.age, "age")), expected)
```

Reply:
I changed the way to use the base dataframe.